docs/cloud.rst

```diff
@@ -0,0 +1,201 @@
+^^^^^^^^^^^^
+
+Many pre-existing datasets are browseable at `catalog.pangeo.io <http://catalog.pangeo.io/>`_.
+This catalog is pre-configured to make it easy to open the datasets.
+In most cases, it's as simple as::
+
+   from intake import open_catalog
+   cat = open_catalog("https://raw.githubusercontent.com/pangeo-data/pangeo-datastore/master/intake-catalogs/atmosphere.yaml")
+   ds  = cat["gmet_v1"].to_dask()
+
+To open datasets that are not part of an intake catalog, we recommend using
+the `filesystem-spec <https://filesystem-spec.readthedocs.io/en/latest/>`_ package
+and its related packages `gcsfs <https://gcsfs.readthedocs.io/en/latest/>`_
+(for Google Cloud Storage)
+and `s3fs <https://s3fs.readthedocs.io/en/latest/>`_
+(for Amazon S3 and all S3-compatible object stores).
+
+For example, to open a public file from Google Cloud Storage, you would do::
+
+   import pandas as pd
+   import fsspec
+   path = 'gs://cmip6/cmip6-zarr-consolidated-stores.csv'
+   with fsspec.open(path) as f:
+       df = pd.read_csv(f)
+
+Zarr stores can be opened using ``.get_mapper`` methods from fsspec, gscsfs, and s3fs.
+For examples, see
+
+- `Zarr Docs on Distributed Cloud Storage <https://zarr.readthedocs.io/en/stable/tutorial.html?highlight=s3fs#distributed-cloud-storage>`_
+- `Xarray Docs on Cloud Storage <http://xarray.pydata.org/en/stable/io.html#cloud-storage-buckets>`_
+
+
+Writing Data
+^^^^^^^^^^^^
+
+Writing data (and reading private data) requires credentials for authentication.
+Pangeo Cloud does not provide credentials to individual users.
+Instead you must sign up for your own account with the cloud provider and manage your own storage.
+(Most cloud providers offer several hundred dollars worth of free credits for new accounts.)
+
+On S3-type storage, you will have a client key and client secret associated with you account.
+The following code creates a writeable filesystem::
+
+   fs = s3fs.S3FileSystem(key='<YOUR_CLIENT_KEY>', secret='<YOUR_CLIENT_SECRET')
+
+Non-AWS S3 services (e.g. Wasabi Cloud) can be configured by passing an argument
+such as ``client_kwargs={'endpoint_url': 'https://s3.us-east-2.wasabisys.com'}``
+to ``S3FileSystem``.
+
+For Google Cloud Storage, the best practice is to create a
+`service account <https://cloud.google.com/iam/docs/service-accounts>`_ with
+appropriate permissions to read / write your private bucket.
+You upload your service account key (a .json file) to your Pangeo Cloud
+home directory and then use it as follows::
+
+   import json
+   import gcsfs
+   with open('<your_token_file>.json') as token_file:
+       token = json.load(token_file)
+   gcs = gcsfs.GCSFileSystem(token=token)
+
+You can then read / write private files with the ``gcs`` object.
+
+Scratch Bucket
+^^^^^^^^^^^^^^
+
+Pangeo Cloud environments are configured with a "scratch bucket," which
+allows you to temporarily store data. Credentials to write to the scratch
+bucket are pre-loaded into your Pangeo Cloud environment.
```

Comment:
@TomAugspurger - can we specify here the environment variable that points to the user's scratch path?

Anything else to add?

Reply:
Dunno yet about the environment variable: https://github.com/pangeo-data/pangeo-cloud-federation/issues/610#issuecomment-651283370

So perhaps just an example spelling things out?

```python
ds.to_zarr("gcs://pangeo-scratch/<your-username>/data.zarr")
```

@scottyhq what did you call the scratch bucket for S3? Are you comfortable with including it here in the pangeo docs?

Reply:
The AWS bucket is also called pangeo-scratch. I'll get to adding this to the docker images today. It'll be called `PANGEO_SCRATCH` and point to `<protocol>://pangeo-scratch/<jupyterhub-username>/`